torch/testing/_internal/common_methods_invocations.py

```diff
@@ -1429,6 +1429,39 @@ def _make_tensor_helper(shape, low=None, high=None):
 
     return samples
 
+
+def reference_sign(x):
+    if x.dtype == np.bool_:
+        # `np.sign` doesn't support `bool`.
+        # >>> np.sign(True)
+        # ufunc 'sign' did not contain a loop
+        # with signature matching types dtype('bool') -> dtype('bool')
+        return np.sign(x, dtype=np.uint8)
```

Comment:
should this then copy the result to a boolean array?

Reply:
That makes sense. (though I think it should have failed. Will check how it passed)

Reply:
https://github.com/pytorch/pytorch/blob/527c1e0e37b7c65148bcbc390b65e94fb4624a9d/test/test_unary_ufuncs.py#L299-L300

At the line above expected_dtype is `torch.uint8` while dtype is `torch.bool` . `can_cast` returns False and hence `exact_dtype` is set to False.
Thus the test worked.